### ISSUE TITLE
Unsaved children are not able to be saved when updating parent object

### ISSUE THREAD

**Issue by TylerBrock** [ParsePlatform/Parse-SDK-JS#89]
Adding to our list of items in #73.

In the below code we make two classes `Brand` and `Item`. Let's say that Brands can have many Items and that this relationship is modeled by an array of `Item` pointers set on the "items" field for `Brand`.

We then create a new `Item` and `Brand`, setting the `item` as an array of pointer on `brand`.

It saves just fine. Parse realized that `item` is unsaved and persists it before saving `brand`. It follows up by saving `brand` and resolving the promise.

However, if we then try to add unsaved objects to the already persisted brand it becomes unable to save the unsaved objects giving the error:

> "Cannot create a pointer to an unsaved ParseObject"

**Setup**
```js
'use strict';

var Parse = require('parse/node');

class Brand extends Parse.Object {
  constructor(attributes, options) {
    super('Brand', attributes, options);
  }
}
Parse.Object.registerSubclass('Brand', Brand);

class Item extends Parse.Object {
  constructor(attributes, options) {
    super('Item', attributes, options);
  }
}
Parse.Object.registerSubclass('Item', Item);
```

**Test**
```js
var item = new Item();
var brand = new Brand({items: [item]});
brand.save().then(function(savedBrand) {
  console.log('works fine');

  var item2 = new Item();
  var brandItems = brand.get('items');
  brandItems.push(item2);
  brand.set("items", brandItems);

  // parse should save unsaved items successfully here
  return brand.save().then(function(updatedBrand) {
    console.log('cool');
  });
});
```

This seems like a bug. It works just fine in the 1.5 SDK. Please let me know what you think. Also I'd love to help fix it if you point me in the right direction. Anything I can do to help, let me know.

**Comment by TylerBrock** [comment]
Also I'd love to help fix it if you point me in the right direction. Anything I can do to help, let me know.

**Comment by andrewimm** [comment]
Resolved in 5e86f78f05aea88b42473cafd7bb186fef690921, will be released in 1.6.14
